# プロ野球でわかる！はじめての統計学

In [1]:
import numpy as np
import pandas as pd
import requests
import json
import numpy as np

## 偏差
```
偏差 = 元のデータ - 平均値
```

## 分散
```
分散 = (Σ偏差^2) ÷ データ数
```

## 標準偏差
```
標準偏差 = sqrt(分散)
```

In [2]:
org_data = pd.read_csv('./statistics_data/2章サンプルデータ.csv')

avg_data = org_data[['選手名','打率']]

avg_data = avg_data.astype({'打率': 'float64'})

avg_data['偏差'] = avg_data['打率'] - avg_data.mean()['打率']

avg_data['偏差^2'] = avg_data['偏差'] * avg_data['偏差']

dispersion = avg_data.sum()['偏差^2'] / 185
print(dispersion)

0.0014787297005113222


In [3]:
standard_deviation = np.sqrt(dispersion)

In [4]:
standard_deviation

0.03845425464771515

In [5]:
avg_data

選手名     打率        偏差          偏差^2
0        山田哲人  0.329  0.078005  6.084843e-03
1        川端慎吾  0.336  0.085005  7.225919e-03
2          雄平  0.270  0.019005  3.612054e-04
3        畠山和洋  0.268  0.017005  2.891838e-04
4        中村悠平  0.231 -0.019995  3.997838e-04
5        大引啓次  0.225 -0.025995  6.757189e-04
6        比屋根渉  0.230 -0.020995  4.407730e-04
7        上田剛史  0.263  0.012005  1.441298e-04
8      Ｍ・デニング  0.222 -0.028995  8.406865e-04
9        荒木貴裕  0.253  0.002005  4.021651e-06
10       田中浩康  0.201 -0.049995  2.499459e-03
11       今浪隆博  0.317  0.066005  4.356714e-03
12       森岡良介  0.188 -0.062995  3.968319e-03
13       三輪正義  0.233 -0.017995  3.238054e-04
14       坂本勇人  0.269  0.018005  3.241946e-04
15       長野久義  0.251  0.000005  2.921841e-11
16       亀井善行  0.272  0.021005  4.412271e-04
17      阿部慎之助  0.242 -0.008995  8.090273e-05
18       片岡治大  0.244 -0.006995  4.892435e-05
19       村田修一  0.236 -0.014995  2.248379e-04
20      立岡宗一郎  0.304  0.053005  2.809573e-03
21       井端弘和  0.234 -0.016995  2.888162e-04
22   Ｌ・アンダーソン  0.252  0.001005  1.010840e-06
23       小林誠司  0.226 -0.024995  6.247298e-04
24       高橋由伸  0.278  0.027005  7.292919e-04
25        橋本到  0.219 -0.031995  1.023654e-03
26       大田泰示  0.277  0.026005  6.762811e-04
27       相川亮二  0.313  0.062005  3.844670e-03
28       堂上剛裕  0.276  0.025005  6.252703e-04
29        鳥谷敬  0.281  0.030005  9.003244e-04
..        ...    ...       ...           ...
155       伊藤光  0.271  0.020005  4.002162e-04
156   Ｆ・カラバイヨ  0.252  0.001005  1.010840e-06
157    Ｅ・ヘルマン  0.267  0.016005  2.561730e-04
158      西野真弘  0.304  0.053005  2.809573e-03
159     小谷野栄一  0.295  0.044005  1.936476e-03
160    Ｔ・ブランコ  0.194 -0.056995  3.248384e-03
161      山崎勝己  0.204 -0.046995  2.208492e-03
162      縞田拓弥  0.226 -0.024995  6.247298e-04
163      宮﨑祐樹  0.252  0.001005  1.010840e-06
164       原拓也  0.170 -0.080995  6.560124e-03
165      川端崇義  0.239 -0.011995  1.438703e-04
166      平野恵一  0.262  0.011005  1.211189e-04
167      坂口智隆  0.262  0.011005  1.211189e-04
168      岩崎恭平  0.281  0.030005  9.003244e-04
169     松井稼頭央  0.256  0.005005  2.505408e-05
170    Ｗ・ペーニャ  0.268  0.017005  2.891838e-04
171      藤田一也  0.270  0.019005  3.612054e-04
172      後藤光尊  0.237 -0.013995  1.958487e-04
173       嶋基宏  0.219 -0.031995  1.023654e-03
174        銀次  0.301  0.050005  2.500541e-03
175   Ｚ・ウィーラー  0.255  0.004005  1.604327e-05
176       聖澤諒  0.252  0.001005  1.010840e-06
177   Ｇ・サンチェス  0.226 -0.024995  6.247298e-04
178      福田将儀  0.216 -0.034995  1.224622e-03
179      中川大志  0.235 -0.015995  2.558271e-04
180      西田哲朗  0.220 -0.030995  9.606649e-04
181      牧田明久  0.311  0.060005  3.600649e-03
182      岡島豪郎  0.168 -0.082995  6.888103e-03
183      伊志嶺忠  0.241 -0.009995  9.989192e-05
184      阿部俊人  0.196 -0.054995  3.024405e-03

[185 rows x 4 columns]

In [6]:
avg_data.mean()['打率']

0.25099459459459456

In [7]:
min_range = avg_data.mean()['打率'] - standard_deviation

In [8]:
max_range = avg_data.mean()['打率'] + standard_deviation

In [9]:
min_range

0.2125403399468794

In [10]:
max_range

0.28944884924230974

In [11]:
avg_data_2 = org_data[['選手名','打率']]

In [12]:
avg_data_2 = avg_data_2.astype({'打率': 'float64'})

In [13]:
avg_data_2.sort_values('打率', ascending=False, inplace=True)

In [16]:
avg_data_2

選手名     打率
0       柳田悠岐  0.363
1       秋山翔吾  0.359
2       川端慎吾  0.336
3       山田哲人  0.329
4       近藤健介  0.326
5       清田育宏  0.317
6       筒香嘉智  0.317
7       今浪隆博  0.317
8       相川亮二  0.313
9       牧田明久  0.311
10    Ｒ・ナニータ  0.308
11     立岡宗一郎  0.304
12      西野真弘  0.304
13        銀次  0.301
14       中村晃  0.300
15       荒波翔  0.298
16      和田一浩  0.298
17     小谷野栄一  0.295
18      藤井淳志  0.295
19      杉谷拳士  0.295
20      脇谷亮太  0.294
21      角中勝也  0.293
22      Ｈ・ルナ  0.292
23     Ｊ・ロペス  0.291
24     伊志嶺翔大  0.291
25      宮﨑敏郎  0.289
26      今江敏晃  0.287
27      松田宣浩  0.287
28       森友哉  0.287
29      淺間大基  0.285
..       ...    ...
155      嶋基宏  0.219
156     吉村裕基  0.217
157     福田将儀  0.216
158     江越大賀  0.214
159    炭谷銀仁朗  0.211
160  Ｊ・ハーミッダ  0.211
161     高橋周平  0.208
162     倉本寿彦  0.208
163     山崎勝己  0.204
164     大谷翔平  0.202
165     髙城俊人  0.202
166     田中浩康  0.201
167     阿部俊人  0.196
168     鬼﨑裕司  0.196
169     鶴岡慎也  0.195
170     大野奨太  0.194
171   Ｔ・ブランコ  0.194
172     新井良太  0.193
173     森岡良介  0.188
174     外崎修汰  0.186
175     桑原将志  0.184
176     杉山翔大  0.183
177    黒羽根利規  0.178
178     髙谷裕亮  0.175
179     藤井彰人  0.172
180     田村龍弘  0.170
181      原拓也  0.170
182     岡島豪郎  0.168
183     松井雅人  0.135
184      細川亨  0.116

[185 rows x 2 columns]

In [15]:
avg_data_2.reset_index(drop=True, inplace=True)

## 標準化得点
標準化得点 = (元のデータ - 平均値) ÷ 標準偏差

### 例
2015山田哲人の標準化得点

- (0.329 - 0.251) ÷ 0.038 = 2.03

## 偏差値
```
標準化得点 = (元のデータ - 平均値) ÷ 標準偏差 × 10 + 50
```

### 例
2015山田哲人の打率偏差値

- (0.329 - 0.251) ÷ 0.038 × 10 + 50 = 70.3

In [20]:
testes = (0.329 - avg_data.mean()['打率']) / standard_deviation * 10 + 50
print(testes)

70.28524700843222


## 非線型変換
GB/FBみたいな、比率のような間隔が等しくないものを比較する際に元の値を変換する手法
```
log10(元のデータ)
```

### 箱ひげ図
最小値と最大値、四分位偏差と中央値を一気に確認できるグラフ（メモ）

### グラフの最大値と最小値
大体集めたデータの平均値±2標準偏差の値をそれぞれ最大値、最小値とするといい（らしい）